components/brave_wallet/browser/meld_integration_response_parser.cc

```diff
@@ -0,0 +1,408 @@
+#include "tools/json_schema_compiler/util.h"
+
+namespace {
+
+std::optional<brave_wallet::mojom::MeldLogoImagesPtr> ParseMeldLogos(
+    const std::optional<brave_wallet::meld_integration_responses::Logos>&
+        logos) {
+  if (!logos) {
+    return std::nullopt;
+  }
+
+  auto logo_images = brave_wallet::mojom::MeldLogoImages::New(
+      logos->dark, logos->dark_short, logos->light, logos->light_short);
+
+  return std::move(logo_images);
+}
+
+std::optional<std::vector<brave_wallet::mojom::MeldRegionPtr>> ParseMeldRegions(
+    const std::optional<
+        std::vector<brave_wallet::meld_integration_responses::Region>>&
+        regions) {
+  if (!regions) {
+    return std::nullopt;
+  }
+
+  std::vector<brave_wallet::mojom::MeldRegionPtr> result;
+  for (const auto& region_value : *regions) {
+    auto reg = brave_wallet::mojom::MeldRegion::New(region_value.region_code,
+                                                    region_value.name);
+    result.emplace_back(std::move(reg));
+  }
+
+  if (result.empty()) {
+    return std::nullopt;
+  }
+
+  return result;
+}
+
+std::optional<base::flat_map<std::string, std::string>>
+ParseOptionalMapOfStrings(const std::optional<base::Value>& val) {
+  if (!val || !val->is_dict()) {
+    return std::nullopt;
+  }
+
+  base::flat_map<std::string, std::string> result;
+  for (const auto [key, value] : val->GetDict()) {
+    result.insert_or_assign(key, value.GetString());
+  }
+  if (result.empty()) {
+    return std::nullopt;
+  }
+
+  return result;
+}
+
+}  //  namespace
+
+namespace brave_wallet {
+
+std::optional<std::vector<mojom::MeldServiceProviderPtr>> ParseServiceProviders(
+    const base::Value& json_value) {
+  // Parses results like this:
+  // {
+  //    "categories": [ "CRYPTO_ONRAMP" ],
+  //    "categoryStatuses": {
+  //       "CRYPTO_ONRAMP": "LIVE"
+  //    },
+  //    "logos": {
+  //       "dark": "https://images-serviceprovider.meld.io/BANXA/logo_dark.png",
+  //       "darkShort":
+  //       "https://images-serviceprovider.meld.io/BANXA/short_logo_dark.png",
+  //       "light":
+  //       "https://images-serviceprovider.meld.io/BANXA/logo_light.png",
+  //       "lightShort":
+  //       "https://images-serviceprovider.meld.io/BANXA/short_logo_light.png"
+  //    },
+  //    "name": "Banxa",
+  //    "serviceProvider": "BANXA",
+  //    "status": "LIVE",
+  //    "websiteUrl": "http://www.banxa.com"
+  // }
+
+  if (!json_value.is_list()) {
+    LOG(ERROR) << "Invalid response, could not parse JSON, JSON is not a list";
+    return std::nullopt;
+  }
+
+  std::vector<mojom::MeldServiceProviderPtr> service_providers;
+  for (const auto& sp_item : json_value.GetList()) {
+    const auto service_provider_value =
+        meld_integration_responses::ServiceProvider::FromValue(sp_item);
+    if (!service_provider_value) {
+      LOG(ERROR)
+          << "Invalid response, could not parse JSON, JSON is not a dict";
+      return std::nullopt;
```

Comment:
I would say we don't need LOGs in this file as they most likely would be never seen by anyone